src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService_FindDeclaration.cs

```diff
@@ -164,16 +172,20 @@ class NestedType
                 }
             }
 
-            // Generate into any decl we can find.
+            // If there is a declaration in a non auto-generated file, prefer it.
+            Func<Document,bool> isGeneratedCode = 
+                solution.Workspace.Services.GetService<IGeneratedCodeRecognitionService>().IsGeneratedCode;
```

Comment:
Side note, you do edit the Dispose method in the Designer file for Forms/Controls.